src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,77 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Collections;
+
 public class WorkWithFile {
+    static final int INDEX_ZERO = 0;
+    static final int INDEX_ONE = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        ArrayList<String> strings = new ArrayList<>();
+        readFile(strings, fromFileName);
+        generateReport(strings);
+        writeToFile(strings, toFileName);
+    }
+
+    public void readFile(ArrayList<String> array, String fileName) {
+        try {
+            try (InputStream inputStream = new FileInputStream(fileName);
+                    BufferedReader bufferedReader = new BufferedReader(
+                            new InputStreamReader(inputStream))) {
+                while (bufferedReader.ready()) {
+                    array.add(bufferedReader.readLine());
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file: " + fileName, e);
+        }
+    }
+
+    public void generateReport(ArrayList<String> array) {
+        for (int i = 0; i < array.size(); i++) {
+            for (int j = i + 1; j < array.size(); j++) {
+                String[] currentStrings = array.get(i).split(",");
+                String[] nextString = array.get(j).split(",");
+                if (currentStrings[INDEX_ZERO].equals(nextString[INDEX_ZERO])) {
+                    int sum = (Integer.parseInt(currentStrings[INDEX_ONE]))
+                            + (Integer.parseInt(nextString[1]));
+                    array.set(i, currentStrings[INDEX_ZERO] + "," + sum);
+                    array.remove(j);
+                    j--;
+                }
+            }
+        }
```

Comment:
it's hard to understand what this code does. refactor it, please. the goal here is to count the supply amount and buy amount. create two variables for the result add fill them by iterating over array once